app/frontend/types.ts

```diff
@@ -121,3 +123,21 @@ export const enum TxType {
   DELEGATE,
   WITHDRAW,
 }
+
+export type Stakepool = {
+  pledge: string
+  margin: number
+  fixedCost: string
+  url: string
+  name: string
+  ticker: string
+  homepage: string
+  poolHash: string
+  validationError?: any
+}
+
+export type Stakepools = {
```

Comment:
the name is dangerously close to `Stakepool` and may seem to be some simple collection of them rather than an object with such non-trivial interface. IMHO a name `StakepoolsView` (borrowing the term "view" from databases) would be more descriptive and less missleading

Reply:
https://www.tutorialsteacher.com/typescript/typescript-interface

Reply:
Refactored